Tests/SwaggerParserTests/SchemaStructureTests.swift

```diff
@@ -0,0 +1,35 @@
+import XCTest
+@testable import SwaggerParser
+
+class SchemaStructureTests: XCTestCase {
+    func testReferenceMetadata() throws {
+        let jsonString = try fixture(named: "test_reference_metadata.json")
+        let swagger = try Swagger(JSONString: jsonString)
+        
+        guard
+            let fooDefinition = swagger.definitions.first(where: { $0.name == "Foo" }),
+            case .object(let foo) = fooDefinition.structure else
+        {
+            return XCTFail("Foo is not an object schema.")
+        }
+        
+        guard
+            let barProperty = foo.properties["bar"],
+            case .structure(let barReference) = barProperty,
+            case .object(let bar) = barReference.schema else
+        {
+            return XCTFail("Bar property is not a reference to an object schema.")
+        }
+        
+        guard
+            let bazProperty = bar.properties["baz"],
+            case .string = bazProperty else
```

Comment:
If you're not unwrapping the variable I think you can just check:
```
guard let bazProperty = bar.properties["baz"], bazProperty == .string else {
```
In either case I think that fits into one line. I set my page guide at 110 characters.